### DISCUSSION TITLE
Sync with OpenAIP's OpenAIR export format

### DISCUSSION THREAD

**Discussion by reskume** [naviter/seeyou_file_formats#4] [General]
Hi there!

Thanks for your effort to document and specify the OpenAir format used in Naviter products publicly! I recently had a chat with Luka on the Aero about the various formats that exist in the wild and that it would simply benefit the whole soaring community if we can settle on a single format spec.

I took a look at your spec and so far, I only think an "UNCLASSIFIED" should be added to the ICAO class attribute values when something like restricted areas are described. Afaik, they (and some others) do not have a defined ICAO class.
Another notable difference between the Naviter and the openAIP OpenAir files is the allowed AY types list. We directly transfer the internally used types into the AY command and dump all "known" types as reference in the files doc header.
I fully understand that a hardware vendor has to restrict the list of allowed types  for their system and thus I think keeping both formats in sync doesn't make much sense. I suppose the best way to handle this is to provide the OpenAIP users with the possibility to export data into the "Naviter OpenAir format" alongside the other OpenAIP formats as well. I'll start adding this feature as soon as the format is deemed ready - or is this already stable and can be implemented?

For completeness, I leave a list of all used OpenAIP types here, maybe some can be added to your list as well:

```yaml
UNCLASSIFIED: 0
RESTRICTED: 1
DANGER: 2
PROHIBITED: 3
# Controlled Tower Region
CTR: 4
# Transponder Mandatory Zone
TMZ: 5
# Radio Mandatory Zone
RMZ: 6
# Terminal Maneuvering Area
TMA: 7
# Temporary Reserved Area
TRA: 8
# Temporary Segregated Area
TSA: 9
# Flight Information Region
FIR: 10
# Upper Flight Information Region
UIR: 11
# Air Defense Identification Zone
ADIZ: 12
# Airport Traffic Zone
ATZ: 13
# Military Airport Traffic Zone
MATZ: 14
# Airway
AWY: 15
# Military Training Route
MTR: 16
# Alert Area
ALERT: 17
# Warning Area
WARNING: 18
# Protected airspace
PROTECTED: 19
# Helicopter Traffic Zone
HTZ: 20
# Gliding Sector
GLIDING_SECTOR: 21
# Transponder Setting
TRP: 22
# Traffic Information Zone (can be handled like a RMZ)
TIZ: 23
# Traffic Information Area (can be handled like a RMZ)
TIA: 24
# Military Training Area
MTA: 25
# Controlled Traffic Area
CTA: 26
# ACC Sector -> Airspace providing available communication frequency in very remote areas
ACC_SECTOR: 27
AERIAL_SPORTING_RECREATIONAL: 28
OVERFLIGHT_RESTRICTION: 29
# MILITARY ROUTE
MRT: 30
# TSA/TRA FEEDING ROUTE - an air route allowing aircraft to access a TSA or TRA and transit between those areas
TFR: 31
# VFR Sector - Used in e.g. Italy. Intention is (in Italy) to make VFR traffic aware of the fact
# that they may interfere with IFR traffic in the sector.
VFR_SECTOR: 32
FIS_SECTOR: 33
# airspace mainly used in Frane to allow VFR traffic in CTAs (Lower traffic area)
LTA: 34
# airspace mainly used in Frane to allow VFR traffic in CTAs (Upper traffic area)
UTA: 35
`

**Comment by akolarck** [comment]
@reskume This is very helpful indeed, thank you.

Can you elaborate on how some of these types are used? I am specifically interested in

`UNCLASSIFIED`
This relates to the ICAO Class of airspace and not type - correct? Or do some in your community also use an "unclassified airspace type" as a definition?

```
# Protected airspace
PROTECTED: 19
```
Is this used for parks and wildlife protection areas or something else?

```
# Transponder Setting
TRP: 22
```
Should we make an extra field out of this, similar to AF (Airfield frequency)? It would theoretically enable us to communicate to a Transponder to set the squawk similar to what can be done with a radio. 

`OVERFLIGHT_RESTRICTION: 29`
How is this used and how is this different from a "Restricted" airspace type?

**Comment by reskume** [comment]
Now that you mention it, we have the `UNCLASSIFIED` value in both the ICAO class and the type values. There are combinations when we set them, for example a `RESTRICTED` airspace will have the ICAO class set to `UNCLASSIFIED`. We use the `UNCLASSIFIED` also as fallback if the airspace has no type defined at all, e.g. class E/G. But those cases are very rare.

Both the `OVERFLIGHT_RESTRICTION` and `PROTECTED` could be used to label the same airspace. But we currently have a pending request to add some non-official airspaces in the US that can best be described with "compliance requested". For them and for explicitly protected areas like parks (also official point in the AIP) we introduced this additional type `OVERFLIGHT_RESTRICTION`. But I agree, both can be used interchangeably - it simply adds some more information.

We already also thought about adding a transponder setting field to our OpenAIP airspace model since this will be more and more important in the future when entering predefined areas like wave windows etc. I would also definitely vote for a new optional field for the the OpenAIR format!

**Comment by llauner** [comment]
Hello
Your proposal for the OpenAir extended format looks good. I think all of the necessary types are there.
Maybe, just as a suggestion, the "short" version for some of the tags could be supported as well ?
AERIAL_SPORTING_RECREATIONAL = ASR
GLIDING_SECTOR = GS
...

Regards,
Ludovic

**Comment by lordfolken** [comment]
Hi There, 
I'm just happy that we have a place to define the standard, and hopefully also develop it further. 
I'll try to align XCSoar's implementation as much as possible.

**Comment by akolarck** [comment]
@reskume We are in the process of supporting all the above airspace types. Can you please provide a few examples and explanations:

1. How is the TRP used as an airspace type ("AY" I suppose)
2. How is ACC_SECTOR used as an airspace type
3. TSA/TRA FEEDING ROUTE - an air route allowing aircraft to access a TSA or TRA and transit between those areas: TFR: 31
TFR should be reserved for "Temporary Flight Restriction" which is widely used in the USA for fires and airspace that protects VIP movements. Please confirm that you can change this on your side.

Thanks.

**Comment by lsalehar** [comment]
Hi All,

I just wanted to let everyone know that the version V2.1 of the OpenAIR format changes the `AA NONE/NONE` to AA NONE, for easier writing of the files and it makes more sense syntactically. Version 2 of the format supports all of the OpenAIP types. I will close this discussion for now. We can discuss other features of the format as we need fit.

Thank you all for participating!
